src/Analyzers/CSharp/CodeFixes/UseCollectionInitializer/CSharpUseCollectionInitializerCodeFixProvider.cs

```diff
@@ -27,6 +32,7 @@ internal class CSharpUseCollectionInitializerCodeFixProvider :
             MemberAccessExpressionSyntax,
             InvocationExpressionSyntax,
             ExpressionStatementSyntax,
+            ForEachStatementSyntax,
```

Comment:
new capability.  if we see a `foreach` of the right shape, we can convert that to `.. expr` if we're creating a collection expression at the end of the day.